DataManagementSystem/Utilities/DMSHelpers.py

```diff
@@ -0,0 +1,229 @@
+        return result
+      sites = result['Value']
+      siteSet.update( sites )
+      for site in sites:
+        candidateSEs = gConfig.getValue( '/Resources/Sites/%s/%s/SE' % ( grid, site ), [] )
+        if candidateSEs:
+          siteSEMapping[LOCAL].setdefault( site, set() ).update( candidateSEs )
+          storageElementSet.update( candidateSEs )
+        else:
+          gLogger.debug( 'No SEs defined for site %s' % site )
+
+    # Add Sites from the SiteSEMappingByProtocol in the CS
+    siteSEMapping[PROTOCOL] = {}
+    cfgLocalSEPath = cfgPath( 'SiteSEMappingByProtocol' )
+    result = self.__opsHelper.getOptionsDict( cfgLocalSEPath )
+    if result['OK']:
+      sites = result['Value']
+      for site in sites:
+        candidates = self.__opsHelper.getValue( cfgPath( cfgLocalSEPath, site ), [] )
+        ses = set( candidates )
+        # If a candidate is a site, then all local SEs are eligible
+        for candidate in ses & siteSet:
+          ses.remove( candidate )
+          ses.update( siteSEMapping[LOCAL][candidate] )
+        siteSEMapping[PROTOCOL].setdefault( site, set() ).update( ses )
+
+    # Add Sites from the SiteSEMappingByDownload in the CS, else SiteLocalSEMapping (old convention)
+    siteSEMapping[DOWNLOAD] = {}
+    cfgLocalSEPath = cfgPath( 'SiteSEMappingByDownload' )
+    result = self.__opsHelper.getOptionsDict( cfgLocalSEPath )
+    if not result['OK']:
+      cfgLocalSEPath = cfgPath( 'SiteLocalSEMapping' )
+      result = self.__opsHelper.getOptionsDict( cfgLocalSEPath )
+    if result['OK']:
+      sites = result['Value']
+      for site in sites:
+        candidates = self.__opsHelper.getValue( cfgPath( cfgLocalSEPath, site ), [] )
+        ses = set( candidates )
+        for candidate in ses & siteSet:
+          ses.remove( candidate )
+          ses.update( siteSEMapping[LOCAL][candidate] )
+        siteSEMapping[ DOWNLOAD].setdefault( site, set() ).update( ses )
+
+    self.siteSEMapping = siteSEMapping
+    self.storageElementSet = storageElementSet
+    self.siteSet = siteSet
+    return S_OK( siteSEMapping )
+
+
+  def isSEFailover( self, storageElement ):
+    seList = resolveSEGroup( self.__opsHelper.getValue( 'DataManagement/SEsUsedForFailover', [] ) )
+    return storageElement in resolveSEGroup( seList )
+
+  def isSEForJobs( self, storageElement, checkSE = True ):
+    if checkSE:
+      self.getSiteSEMapping()
+      if storageElement not in self.storageElementSet:
+        return False
+    seList = resolveSEGroup( self.__opsHelper.getValue( 'DataManagement/SEsNotToBeUsedForJobs', [] ) )
+    return storageElement not in resolveSEGroup( seList )
+
+  def isSEArchive( self, storageElement ):
+    seList = resolveSEGroup( self.__opsHelper.getValue( 'DataManagement/SEsUsedForArchive', [] ) )
+    return storageElement in resolveSEGroup( seList )
+
+  def getSitesForSE( self, storageElement, connectionLevel = None ):
```

Comment:
At the beginning of the module you set 
LOCAL = 1
PROTOCOL = 2
DOWNLOAD = 3
but in this method you request in input only a string. What if you get an integer?